assets/js/app/DataLayers/scatter.js

```diff
@@ -466,3 +467,107 @@ LocusZoom.DataLayers.add("scatter", function(layout){
+                var bv = bk.toString ? bk.toString().toLowerCase() : bk;
+                return (av === bv) ? 0 : (av < bv ? -1 : 1);});
+        sourceData.forEach(function(d, i){
+            // Implementation detail: Scatter plot requires specifying an x-axis value, and most datasources do not
+            //   specify plotting positions. If a point is missing this field, fill in a synthetic value.
+            d[xField] = d[xField] || i;
+        });
+        return sourceData;
+    },
+
+    /**
+     * Identify the unique categories on the plot, and update the layout with an appropriate color scheme
+     *
+     * Also identify the min and max x value associated with the category, which will be used to generate ticks
+     * @private
+     * @returns {Object.<String, Number[]>} Series of entries used to build category name ticks {category_name: [min_x, max_x]}
+     */
+    _generateCategoryBounds: function() {
+        // TODO: API may return null values in category_field; should we add placeholder category label?
+        // The (namespaced) field from `this.data` that will be used to assign datapoints to a given category & color
+        var category_field = this.layout.x_axis.category_field;
+        var xField = this.layout.x_axis.field || "x";
+        var uniqueCategories = {};
+        this.data.forEach(function(item) {
+            var category = item[category_field];
+            var x = item[xField];
+            var bounds = uniqueCategories[category] || [x, x];
+            uniqueCategories[category] = [Math.min(bounds[0], x), Math.max(bounds[1], x)];
+        });
+
+        var categoryNames = Object.keys(uniqueCategories);
+        // Construct a color scale with a sufficient number of visually distinct colors
+        // TODO: This will break for more than 20 categories in a single API response payload for a single PheWAS plot
+        var color_scale = categoryNames.length <= 10 ? d3.scale.category10 : d3.scale.category20;
+        var colors = color_scale().range().slice(0, categoryNames.length);  // List of hex values, should be of same length as categories array
+
+        this.layout.color.parameters.categories = categoryNames;
+        this.layout.color.parameters.values = colors;
+        return uniqueCategories;
+    },
+
+    getTicks: function(dimension) { // Overrides parent method
+        if (["x", "y"].indexOf(dimension) === -1) {
+            throw("Invalid dimension identifier");
+        }
+        var categoryBounds = this.categories;
+        if (!categoryBounds  || !Object.keys(categoryBounds).length) {
+            return [];
+        }
+
+        return Object.keys(categoryBounds).map(function(category) {
+            var bounds = categoryBounds[category];
+            var diff = bounds[1] - bounds[0];
+            var center = bounds[0] + (diff !== 0 ? diff: bounds[0]) / 2;  // Center tick under one or many elements as appropriate
```

Comment:
Unless someone else had suggested PheWAS x ticks be centered on their extents and had a good reason to back it up I'd say this centering should be optional (driven by the axis layout) and default off. Compare what's showing in this branch currently vs. [existing LocusZoom PheWAS implementations](https://www.michigangenomics.org/genetic_data.html). It certainly makes sense sometimes to center a tick on its extent, but for PheWAS since the phenotype names can get long we went for the 45° rotation to help everything fit and with that it's a lot easier to track categories to their associated scatter points when the text anchor is aligned with the start of the extent.

Reply:
I've made the change suggested- good point about long category names. This is now configurable.

The panel can now pass special configuration parameters to `dataLayer.getTicks()` to control things that can't be overridden after the fact; category_scatter will recognize `position` values of `left|center|right`.

Colors are now shared between tick points and categories too. And as a final additional layer of reinforcement, the category name is part of the tooltips for each individual point, even in the default layout.